pkg/destroy/azure/azure.go

```diff
@@ -0,0 +1,247 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/dns/mgmt/2018-03-01-preview/dns"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	session "github.com/openshift/installer/pkg/asset/installconfig/azure"
+	"github.com/openshift/installer/pkg/destroy"
+	"github.com/openshift/installer/pkg/types"
+)
+
+// ClusterUninstaller holds the various options for the cluster we want to delete.
+type ClusterUninstaller struct {
+	SubscriptionID string
+	Authorizer     autorest.Authorizer
+
+	InfraID string
+
+	Logger logrus.FieldLogger
+
+	resourceGroupsClient resources.GroupsGroupClient
+	zonesClient          dns.ZonesClient
+	recordsClient        dns.RecordSetsClient
+}
+
+func (o *ClusterUninstaller) configureClients() {
+	resourceGroupsClient := resources.NewGroupsGroupClient(o.SubscriptionID)
+	resourceGroupsClient.Authorizer = o.Authorizer
+	o.resourceGroupsClient = resourceGroupsClient
```

Comment:
Why the local variable vs. `o.resourceGroupsClient = resources.NewGroupsGroupClient(o.SubscriptionID)`?  Are we casting to an `Interface` or some such and loosing access to the `Authorizer` property.

Reply:
Azure sdk doesn't have a way to initialize a client with auth.
So the
```
client := NewClient()
client.Authorizer = auth
o.client = client
```
seems more appropriate of create a new client, and use that to initialize/assign _our_ client.

Do you have strongly about not using the local var? If not personally current make more sense to me and i would like to keep it as is...?

Reply:
I was thinking:

```go
o.client = NewClient()
o.client.Authorizer = auth
```

but don't feel strongly either way.

Reply:
If it functions properly, I think Trevor's recommendation is more concise.

Reply:
Sure, fixed with https://github.com/openshift/installer/compare/1bd15ad1005263bf9a9c926eb42c927fca4de33f...9a57507a89a576187db04dd8b9b8f5674c750b80